### PR TITLE
Prevent repeatedly updating previews in specviz2d spectral extraction

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request is an alternative to #2861 to address repeated calls to update spectral extraction previews (especially for "expensive" trace calls) in specviz2d.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py [hunk 1]
```diff
@@ -417,6 +417,7 @@ def update_marks(self, step=None):
                 raise ValueError("step must be one of: trace, bg, ext")
 
     # also listens to is_active from any _interaction_in_*_step methods
```
Comment (rosteen) [inline jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py:419]: Is this comment no longer accurate?

**Comment by rosteen** [comment]
Looks like this also works, I'm open to merging this instead of mine if you prefer.

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py:419]: yes, I think so, good catch!

**Review by rosteen — APPROVED** [review]
100% patch coverage, lowered project coverage? Go home Codecov you're crazy.

**Review by gibsongreen — APPROVED** [review]
Small diff and seems to do the trick for me!

**Pull request merged**